src/Query/Builder.php

```diff
@@ -1034,47 +1034,56 @@ protected function compileWhereBasic(array $where): array
     {
         extract($where);
 
-        // Replace like or not like with a Regex instance.
+        // Replace like with a Regex instance.
         if (in_array($operator, ['like', 'not like'])) {
-            if ($operator === 'not like') {
-                $operator = 'not';
-            } else {
-                $operator = '=';
-            }
-
-            // Convert to regular expression.
-            $regex = preg_replace('#(^|[^\\\])%#', '$1.*', preg_quote($value));
-
-            // Convert like to regular expression.
-            if (! Str::startsWith($value, '%')) {
-                $regex = '^'.$regex;
-            }
-            if (! Str::endsWith($value, '%')) {
-                $regex .= '$';
-            }
+            $regex = preg_replace(
+                [
+                    // Unescaped % are converted to .*
+                    // Group consecutive %
+                    '#(^|[^\\\])%+#',
```

Comment:
Similar to my comment about looking for unescaped delimiters, note that this may catch unescaped wildcards if they follow a backslash which itself had to be escaped. I'm not sure we'll want to care about this, but wanted to point it out nonetheless.

Reply:
Oh, that's one more small step. I'll fix that.

Reply:
The regex is getting complex. https://regex101.com/r/fkTvOe/1

Reply:
Yeah, that doesn't look like something we'll want to maintain. I'm fine leaving this as-is.

Reply:
If anyone has this use-case, they can switch to `Regex`.

Reply:
Yeah, that’s what I would suggest as well.